pkg/leaderelection/leaderelection.go

```diff
@@ -174,15 +184,18 @@ func IsLeader(ctx context.Context, etcdConnectionConfig *brtypes.EtcdConnectionC
 	response, err := client.Status(ctx, endPoint)
 	if err != nil {
 		logger.Errorf("failed to get status of etcd endPoint: %v with error: %v", endPoint, err)
-		return false, err
+		return false, false, err
 	}
 
 	if response.Header.MemberId == response.Leader {
-		return true, nil
+		return true, false, nil
 	} else if response.Leader == NoLeaderState {
-		return false, &errors.EtcdError{
+		return false, false, &errors.EtcdError{
```

Comment:
Is it not possible that a member reports `NoLeaderState` and can still be a learner?

Reply:
`NoLeaderState ` represents that there is no quorum present in the cluster.
so, checking whether the leaner is present or not doesn't make sense as we won't be able to perform any action on learner until there will be a leader in a cluster.
so, IMO it is better to wait until there is a leader as it is in a leader-elector loop, so it will recheck and won't backup fire in any case. Hope that's make sense.